html/pfappserver/root/src/views/Configuration/roles/_components/TheList.vue

```diff
@@ -0,0 +1,517 @@
+<template>
+  <b-card no-body>
+    <b-card-header>
+      <div class="float-right">
+        <base-input-toggle-advanced-mode
+          v-model="advancedMode"
+          :disabled="isLoading"
+          label-left
+        />
+      </div>
+      <h4 class="mb-0">
+        {{ $t('Roles') }}
+        <pf-button-help class="ml-1" url="PacketFence_Installation_Guide.html#_introduction_to_role_based_access_control" />
+      </h4>
+    </b-card-header>    
+    <div class="card-body">
+      <transition name="fade" mode="out-in">
+        <div v-if="advancedMode">
+          <b-form @submit.prevent="onSearchAdvanced" @reset.prevent="onSearchReset">
+            <base-search-input-advanced 
+              v-model="conditionAdvanced"
+              :disabled="isLoading"
+              :fields="fields"
+              @reset="onSearchReset"
+              @search="onSearchAdvanced"
+            />
+            <b-container fluid class="text-right mt-3 px-0">
+              <b-button class="mr-1" type="reset" variant="secondary" :disabled="isLoading">{{ $t('Clear') }}</b-button>
+              <base-button-save-search namespace="roles" v-model="conditionAdvanced"
```

Comment:
If we allow the user to have saved searches, we should make them available from the *Basic Mode* to make them more accessible (since they don't appear in the sidebar).